frame/assets/src/types.rs

```diff
@@ -115,13 +119,15 @@ impl<Balance> ExistenceReason<Balance> {
 }
 
 #[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, MaxEncodedLen, TypeInfo)]
-pub struct AssetAccount<Balance, DepositBalance, Extra> {
+pub struct AssetAccount<Balance, DepositBalance, Extra, AccountId> {
 	/// The balance.
 	pub(super) balance: Balance,
 	/// Whether the account is frozen.
 	pub(super) is_frozen: bool,
 	/// The reason for the existence of the account.
 	pub(super) reason: ExistenceReason<DepositBalance>,
+	/// If the account exists because of `ExistenceReason::Deposit`, the depositor.
+	pub(super) depositor: Option<AccountId>,
```

Comment:
```
+ExistenceReason::DepositHeld((Balance, AccountId)),
```
Yeah, in this case there will be lots of duplication of AccountId, since in most cases its the same with the owner of the account.
Not sure if it covers all the cases, but may be new reason `ExistenceReason::Freeze(Balance, AccountId)`

Reply:
The question is whether or not to add the `depositor` field _or_ add `AccountId` into the meaning of a deposit.

Reply:
I think `depositor` fields is more clear

Reply:
added the variant `ExistenceReason::DepositFrom(Balance, AccountId)`